lib/tokenizers/decode_stream.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Tokenizers.DecodeStream do
+  @moduledoc """
+  Implements streaming decoding functionality for tokenizers.
+  """
+
+  @enforce_keys [:resource]
+  defstruct [:resource]
+
+  @type t :: %__MODULE__{
+          resource: reference()
+        }
+
+  @doc """
+  Creates a new decode stream.
+
+  The `skip_special_tokens` option determines whether special tokens should be skipped during decoding.
+  By default, it is set to `false`.
+  """
+  @spec new(boolean()) :: t()
+  def new(skip_special_tokens \\ false) do
+    Tokenizers.Native.decoder_stream_new(skip_special_tokens)
+  end
+
+  @doc """
+  Steps through the decode stream with the given tokenizer and token ID.
+
+  Returns `{:ok, string}` if there's a decoded string, or `{:ok, nil}` if there's nothing more to decode.
+  Returns `{:error, reason}` if an error occurs during decoding.
+  """
+  def step(%__MODULE__{} = decode_stream, tokenizer, id) when is_integer(id) do
+    case Tokenizers.Native.decoder_stream_step(decode_stream, tokenizer, id) do
+      {:ok, result} -> {:ok, result}
+      {:error, reason} -> {:error, reason}
+    end
```

Comment:
```suggestion
    Tokenizers.Native.decoder_stream_step(decode_stream, tokenizer, id)
```

Reply:
Is it actually efficient going step after step by giving specific indexes? Or does the upstream code works better by calling something `.next`? Is this something we should benchmark before?

Reply:
In this PR I translated API as is. Some abstraction can be definitely written on top of it using `Stream`, but it seems to be a part of separate effort.

Decoding large chunks this way looks better for BEAM because control is given back from NIF. There is a possibility to write decoding without dirty, but i'm not sure somebody will invest into it.

Reply:
Got it! Thank you!